src/main/java/core/basesyntax/service/strategy/HandleTransactionStrategyImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.strategy.handlers.BalanceHandler;
+import core.basesyntax.service.strategy.handlers.PurchaseHandler;
+import core.basesyntax.service.strategy.handlers.ReturnHandler;
+import core.basesyntax.service.strategy.handlers.SupplyHandler;
+
+public class HandleTransactionStrategyImpl implements HandleTransactionStrategy {
+    @Override
+    public TransactionHandler get(FruitTransaction.Operation type) {
+        switch (type) {
```

Comment:
[Avoid using switch-cases and if-else constructions. It is recommended to use the Strategy pattern instead](https://mate-academy.github.io/jv-program-common-mistakes/java-core/solid/fruit-shop#avoid-using-switch-cases-and-if-else-constructions-it-is-recommended-to-use-the-strategy-pattern-instead)